src/main/java/com/google/api/codegen/discovery/DiscoveryProviderFactory.java

```diff
@@ -17,11 +17,15 @@
 import com.fasterxml.jackson.databind.JsonNode;
 import com.google.api.Service;
 import com.google.api.codegen.ApiaryConfig;
+import java.io.File;
 import java.util.List;
 
 /** A factory for DiscoveryProviders which perform code generation. */
 public interface DiscoveryProviderFactory {
-  /** Create the provider from the given service, apiaryConfig, sampleConfigOverrides, and id. */
   DiscoveryProvider create(
-      Service service, ApiaryConfig apiaryConfig, List<JsonNode> sampleConfigOverrides, String id);
+      Service service,
+      ApiaryConfig apiaryConfig,
+      List<JsonNode> sampleConfigOverrides,
+      File rubyNamesFile,
```

Comment:
Only Ruby uses a separate file to determine the names of methods/variables. That file is generated by the Ruby client library. Toolkit has a hard reference to a file in `resources`, which requires us to update the repos copy frequently. So passing the file through provides some convenience for me.